.github/workflows/sonar_pr_analysis.yml

```diff
@@ -0,0 +1,80 @@
+---
+name: Sonar analysis for pull requests
+
+on:
+  workflow_run:
+    workflows:
+      - Pull request verification
+    types:
+      - completed
+env:
+  JAVA_DISTRIBUTION: 'temurin'
+  JAVA_VERSION: '17'
+
+# Cancel in-progress sonar branch analysis workflows. We only care about analyzing the latest commit.
+concurrency:
+  group: ${{ github.event.workflow.name }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  branch_sonar_analysis:
+    name: long-lived branch sonar analysis
```

Comment:
This job isn't about long-lived branches, it's about PRs, so we need a different name like:
```
pr_sonar_analysis:
  name: PR sonar analysis
```

Reply:
Done.